src/vm/memory/memory.zig

```diff
@@ -399,6 +427,124 @@ pub fn setUpMemory(memory: *Memory, allocator: Allocator, comptime vals: anytype
     }
 }
 
+test "Memory: validate existing memory" {
+    var allocator = std.testing.allocator;
+
+    var segments = try MemorySegmentManager.init(allocator);
+    defer segments.deinit();
+
+    var builtin = RangeCheckBuiltinRunner.new(8, 8, true);
+    builtin.initializeSegments(segments);
+    try builtin.addValidationRule(segments.memory);
+    var seg = segments.addSegment();
+    _ = seg;
+
+    try setUpMemory(segments.memory, std.testing.allocator, .{
+        .{ .{ 0, 2 }, .{1} },
+        .{ .{ 0, 5 }, .{1} },
+        .{ .{ 0, 7 }, .{1} },
+
+        .{ .{ 1, 1 }, .{1} },
+
+        .{ .{ 2, 2 }, .{1} },
+        .{ .{ 2, 3 }, .{1} },
+        .{ .{ 2, 4 }, .{1} },
+        .{ .{ 2, 5 }, .{1} },
+    });
+    defer segments.memory.deinitData(std.testing.allocator);
+
+    try segments.memory.validateExistingMemory();
+
+    try expect(
+        segments.memory.validated_addresses.contains(Relocatable.new(0, 2)),
+    );
+    try expect(
+        segments.memory.validated_addresses.contains(Relocatable.new(0, 5)),
+    );
+    try expect(
+        segments.memory.validated_addresses.contains(Relocatable.new(0, 7)),
+    );
```

Comment:
Here can we also verify that the other addresses are not in `validated_addresses` because you only added the 0 segment as a validation rule?